stats/opentelemetry/server_metrics.go

```diff
@@ -0,0 +1,200 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package opentelemetry
+
+import (
+	"context"
+	"sync/atomic"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/internal"
+	"google.golang.org/grpc/stats"
+	"google.golang.org/grpc/status"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+)
+
+type serverStatsHandler struct {
+	mo MetricsOptions
+
+	serverMetrics serverMetrics
+}
+
+func (ssh *serverStatsHandler) initializeMetrics() {
+	// Will set no metrics to record, logically making this stats handler a
+	// no-op.
+	if ssh.mo.MeterProvider == nil {
+		return
+	}
+
+	meter := ssh.mo.MeterProvider.Meter("gRPC-Go")
+	if meter == nil {
+		return
+	}
+	setOfMetrics := ssh.mo.Metrics.metrics
+
+	serverMetrics := serverMetrics{}
+	if _, ok := setOfMetrics["grpc.server.call.started"]; ok {
+		scs, err := meter.Int64Counter("grpc.server.call.started", metric.WithUnit("call"), metric.WithDescription("The total number of RPCs started, including those that have not completed."))
+		if err != nil {
+			logger.Errorf("failed to register metric \"grpc.server.call.started\", will not record") // error or log?
+		} else {
+			serverMetrics.callStarted = scs
+		}
+	}
+
+	if _, ok := setOfMetrics["grpc.server.call.sent_total_compressed_message_size"]; ok {
+		ss, err := meter.Int64Histogram("grpc.server.call.sent_total_compressed_message_size", metric.WithUnit("By"), metric.WithDescription("Total bytes (compressed but not encrypted) sent across all response messages (metadata excluded) per RPC; does not include grpc or transport framing bytes."), metric.WithExplicitBucketBoundaries(DefaultSizeBounds...))
+		if err != nil {
+			logger.Errorf("failed to register metric \"grpc.server.call.sent_total_compressed_message_size\", will not record") // error or log?
+		} else {
+			serverMetrics.callSentTotalCompressedMessageSize = ss
+		}
+	}
+
+	if _, ok := setOfMetrics["grpc.server.call.rcvd_total_compressed_message_size"]; ok {
+		sr, err := meter.Int64Histogram("grpc.server.call.rcvd_total_compressed_message_size", metric.WithUnit("By"), metric.WithDescription("Total bytes (compressed but not encrypted) received across all request messages (metadata excluded) per RPC; does not include grpc or transport framing bytes."), metric.WithExplicitBucketBoundaries(DefaultSizeBounds...))
+		if err != nil {
+			logger.Errorf("failed to register metric \"grpc.server.rcvd.sent_total_compressed_message_size\", will not record")
+		} else {
+			serverMetrics.callRcvdTotalCompressedMessageSize = sr
+		}
+	}
+
+	if _, ok := setOfMetrics["grpc.server.call.duration"]; ok {
+		scd, err := meter.Float64Histogram("grpc.server.call.duration", metric.WithUnit("s"), metric.WithDescription("This metric aims to measure the end2end time an RPC takes from the server transport’s perspective."), metric.WithExplicitBucketBoundaries(DefaultLatencyBounds...))
```

Comment:
Description - End-to-end time taken to complete a call from server transport's perspective

Reply:
Done.